src/index.ts

```diff
@@ -93,22 +149,30 @@ class DebuggerHandler<
   async update<W extends ConsolePanel | NotebookPanel | FileEditor>(
     debug: IDebugger,
     widget: W
-  ) {
+  ): Promise<void> {
     const debuggingEnabled = await debug.requestDebuggingEnabled();
-    if (!debug.model || this.handlers[widget.id] || !debuggingEnabled) {
+    if (!debug.model || !debuggingEnabled || !debug.isStarted()) {
+      return;
+    }
+
+    if (this.handlers[widget.id]) {
+      if (!debug.session.debuggedClients.has(debug.session.client.path)) {
```

Comment:
Maybe the two `if` could be combined then?

Reply:
yeah, definitely. Sorry need make break :)